crates/compression/src/ports.rs

```diff
@@ -119,3 +119,29 @@ where
         <D as EvictorDb<T>>::set_latest_assigned_key(self, key)
     }
 }
+
+#[cfg(feature = "fault-proving")]
+pub mod fault_proving {
+    pub trait GetRegistryRoot {
+        type Error: core::fmt::Display;
+        fn registry_root(
+            &self,
+        ) -> Result<crate::compressed_block_payload::v1::RegistryRoot, Self::Error>;
+    }
+
+    impl<D> GetRegistryRoot for &mut D
+    where
+        D: GetRegistryRoot,
+    {
+        type Error = <D as GetRegistryRoot>::Error;
```

Comment:
Do we need to disambiguate here?

```suggestion
        type Error = D::Error;
```

Reply:
addressed in https://github.com/FuelLabs/fuel-core/compare/6771f80153ebf4e0b7a12f4170db26a9751b77f5..4a06900a0d07ff94d9183eb116f4f5deab51b365#diff-872ff25327484ccc3b4211dc4bced653417aac32f3c5ae1369d5610a574a2c3dR136-R141

Reply:
Sweet!